packages/InlineEditing/stories/InlineEditing.stories.mdx

```diff
@@ -0,0 +1,278 @@
+    setIsEditing(true);
+  }
+
+  function handleSubmit({ nextValue }) {
+    setValue(nextValue);
+    setIsEditing(false);
+  }
+
+  // prettier-ignore
+  return (
+    <InlineInput 
+      value={value} 
+      onClick={handleClick} 
+      isEditing={isEditing} 
+      onSubmit={handleSubmit} 
+    />
+  )
+}
+```
+
+  </Block>
+</Blocks>
+
+The previous example is a simple attempt to create an InlineEditng Input. To have it fully functional will still require more work with the keyboard interactions,
+add focus onClick, cancel it onKeyUp === "Escape", etc.
+
+And because wiring a fully functionally InlineEditing component can be tedious and tricky to make it work properly, we are currently providing three components ready to use:
+
+- Input type text
+- TextArea
+- ListBox
+
+## Available InlineEditing Components
+
+### Input
+
+<Blocks>
+  <Block>
+    <InputStory />
+  </Block>
+  <Block>
+
+```jsx
+import Input from "@paprika/inline-editing/lib/Input";
+
+const [isEditing, setIsEditing] = React.useState(false);
+
+<Textarea
+  value="your value"
+  onSubmit={handleSubmit}
+  onEditingClose={() => setIsEditing(false)}
+  onEditingOpen={() => setIsEditing(true)}
+  isEditing={isEditing}
+/>;
+```
+
+  </Block>
+</Blocks>
+
+### TextArea
+
+<Blocks>
+  <Block>
+    <TextAreaStory />
+  </Block>
+  <Block>
+
+```jsx
+import TextArea from "@paprika/inline-editing/lib/Textarea";
+
+const [isEditing, setIsEditing] = React.useState(false);
+
+<Textarea
+  value="your value"
+  onSubmit={handleSubmit}
+  onEditingClose={() => setIsEditing(false)}
+  onEditingOpen={() => setIsEditing(true)}
+  isEditing={isEditing}
+/>;
+```
+
+  </Block>
+</Blocks>
+
+### ListBox
+
+<Blocks>
+  <Block>
+    <ListBoxSingleStory />
+    <div style={{ width: "100%", height: "44px" }}></div>
+    <ListBoxMultipleStory />
+  </Block>
+  <Block>
+
+```jsx
+import ListBox from "@paprika/inline-editing/lib/ListBox";
+
+const [isEditing, setIsEditing] = React.useState(false);
+
+/** 
+  Be aware that ListBox isMulti and the single select option have a different signature for the onChange and onEditing event 
+  isMulti={true} ([indexes], options, selectedIndex, {options})
+  isMulti={false} ([indexes], options, selectedIndex, {options})
+*/
+<ListBox
+  isEditing={isEditing}
+  onChange={() => {}}
+  onClose={() => {}}
+  onEditing={() => {}}
+  onSubmit={handleSubmit}
+  value={value}
+>
+  {someOptions.map(subscription => (
+    <ListBox.Option value={subscription} isSelected={subscription === value} key={subscription}>
+      {subscription}
+    </ListBox.Option>
+  ))}
+</ListBox>;
+```
+
+  </Block>
+</Blocks>
+
+## onSubmit vs onChange
+
+The inline-editing component provides two different callbacks:
+
+- `onChange` Will fire each time the component gets an interaction from the user. (typing, selecting, etc)
+- `onSubmit` Will fire only after the user either clicks `enter` (for certain components) or clicks outside of the editing state (onBlur)
```

Comment:
I'm not sure about the name of this callback – `onSubmit` already has a specific meaning for forms, and this is not that.  Maybe `onFinishEditing`?

Also, does this mean it will fire twice if the user presses `enter` and then `tab`?

Reply:
depending on tab, if tab is exiting the input will fire submit, if its inside of it will don't do anything, submit occurs either by press the key Enter or onblur, Tab doesn't have any play on the trigger.